test/common/stats/symbol_table_impl_test.cc

```diff
@@ -207,11 +207,16 @@ TEST_F(StatNameTest, TestLongSequence) {
 }
 
 TEST_F(StatNameTest, TestUnusualDelimitersRoundtrip) {
-  const std::vector<std::string> stat_names = {".x",   "..x",    "...x",    "foo",     "foo.x",
-                                               ".foo", ".foo.x", ".foo..x", "..foo.x", "..foo..x"};
-  for (auto& stat_name : stat_names) {
-    EXPECT_EQ(stat_name, encodeDecode(stat_name));
-  }
+  EXPECT_EQ("x", encodeDecode(".x"));
+  EXPECT_EQ("x", encodeDecode("..x"));
+  EXPECT_EQ("x", encodeDecode("...x"));
+  EXPECT_EQ("foo", encodeDecode("foo"));
+  EXPECT_EQ("foo.x", encodeDecode("foo.x"));
+  EXPECT_EQ("foo", encodeDecode(".foo"));
+  EXPECT_EQ("foo.x", encodeDecode(".foo.x"));
+  EXPECT_EQ("foo.x", encodeDecode(".foo..x"));
+  EXPECT_EQ("foo.x", encodeDecode("..foo.x"));
+  EXPECT_EQ("foo.x", encodeDecode("..foo..x"));
```

Comment:
I'm not sure why these weren't added in the first place, but can you add some tests with trailing dots also?